File: flutter/lib/domain/constants/use_cases.dart
class UseCases {
  static const List<String> defaults = [
    'formal',
    'informal',
    'party',
    'date',
    'dinner',
    'lunch',
  ];

  static String normalize(String value) {
    return value.trim().toLowerCase();
  }

  static List<String> normalizeList(Iterable<String>? values) {
    if (values == null) return const [];

    final seen = <String>{};
    final normalized = <String>[];

    for (final value in values) {
      final tag = normalize(value);
      if (tag.isEmpty || seen.contains(tag)) continue;
      seen.add(tag);
      normalized.add(tag);
    }

    return normalized;
  }

  static String displayLabel(String value) {
    final normalized = normalize(value);
    if (normalized.isEmpty) return '';
    return '${normalized[0].toUpperCase()}${normalized.substring(1)}';
  }
}

File: frontend/src/lib/use-cases.ts
export const DEFAULT_USE_CASES = [
  'formal',
  'informal',
  'party',
  'date',
  'dinner',
  'lunch',
] as const

export type DefaultUseCase = (typeof DEFAULT_USE_CASES)[number]

export function normalizeUseCase(value: string): string {
  return value.trim().toLowerCase()
}

export function normalizeUseCases(values: string[] | undefined | null): string[] {
  if (!values?.length) return []

  const seen = new Set<string>()
  const normalized: string[] = []

  values.forEach((value) => {
    const tag = normalizeUseCase(value)
    if (!tag || seen.has(tag)) return
    seen.add(tag)
    normalized.push(tag)
  })

  return normalized
}

export function formatUseCaseLabel(value: string): string {
  const normalized = normalizeUseCase(value)
  if (!normalized) return ''
  return normalized.charAt(0).toUpperCase() + normalized.slice(1)
}
